traits/ctraits.c

```diff
@@ -2180,19 +2180,19 @@ call_notifiers(
     PyObject *result, *item, *temp, *args;
     int rc = 0;
 
+    // Do nothing if the user has explicitly requested no traits notifications
+    // to be sent.
+    if ((obj->flags & HASTRAITS_NO_NOTIFY)) {
```

Comment:
Nitpick: can we get rid of the unnecessarily doubly-nested parentheses while we're here?